integration-tests/devtools/src/test/java/io/quarkus/devtools/project/codegen/codestarts/CodestartProjectTest.java

```diff
@@ -0,0 +1,124 @@
+package io.quarkus.devtools.project.codegen.codestarts;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.bootstrap.model.AppArtifactKey;
+import io.quarkus.devtools.PlatformAwareTestBase;
+import io.quarkus.devtools.ProjectTestUtil;
+
+class CodestartProjectTest extends PlatformAwareTestBase {
+
+    private static final Path projectPath = Paths.get("target/codestarts-test");
+
+    @BeforeAll
+    static void setUp() throws IOException {
+        ProjectTestUtil.delete(projectPath.toFile());
+    }
+
+    private Map<String, Object> getTestInputData() {
+        return getTestInputData(null);
+    }
+
+    private Map<String, Object> getTestInputData(final Map<String, Object> override) {
+        final HashMap<String, Object> data = new HashMap<>();
+        data.put("project.version", "1.0.0-codestart");
+        data.put("quarkus.platform.group-id", getPlatformDescriptor().getBomGroupId());
+        data.put("quarkus.platform.artifact-id", getPlatformDescriptor().getBomArtifactId());
+        data.put("quarkus.platform.version", "1.5.2.Final");
+        data.put("quarkus.plugin.group-id", "io.quarkus");
+        data.put("quarkus.plugin.artifact-id", "quarkus-maven-plugin");
+        data.put("quarkus.plugin.version", "1.5.2.Final");
```

Comment:
Is the test actually resolving the 1.5.2.Final platform descriptor?

Reply:
@aloubyansky nop, it's just using those as data in the buildfile. The codestarts code is not resolving anything, it just gets an already resolved platformDescriptor as input.